### ISSUE TITLE
How to actually use created Azure packages

### ISSUE THREAD

**Issue by fenix2222** [davidpeden3/octopus-azure-transforms#1]
Hi David,

I am trying to follow your solution and I understanding all of it, but I just can't figure out how you are actually deploying your azure package? I understand that you re-package cspkg to apply web.config transforms, but wouldn't you then have to:
1.  Add second step (probably powershell) to package it and push it to octopus (basically to itself), so that it creates a new nupkg
2. Add a third step which will perform actual deployment to Azure using the package from Step 2

**Comment by davidpeden3** [comment]
Hey @fenix2222,

First, I followed the Octopus documentation for [deploying to Windows Azure](http://docs.octopusdeploy.com/display/OD/Windows+Azure). Basically, you must install a tentacle on the same machine that hosts the Octopus server. Octopus then deploys locally to the tentacle before uploading your package to azure blob storage. My [deploy.ps1](https://github.com/davidpeden3/octopus-azure-transforms/blob/master/source/Deployment/deploy.ps1) script performs the re-packaging post-transforms (executed by Octopus as configured [here](https://raw.githubusercontent.com/davidpeden3/octopus-azure-transforms/master/documentation/octopus-process.png)) before the re-written package is published to Azure.

In other words, I simply leverage the `deploy.ps1` _hook_ to inject my repackaging step into the native Octopus Azure deployment pipeline.

**Comment by fenix2222** [comment]
Hi David,

Does that mean that this screenshot is actually from "Deploy to Azure" step
type? If so, does that mean that deploy.ps1 and postdeploy.ps1 will run
before actual octopus Azure Deployment script runs in that step? Just
trying to understand how this works

On Wed, May 20, 2015 at 2:43 PM, David Peden notifications@github.com
wrote:

> Hey @fenix2222 https://github.com/fenix2222,
> 
> First, I followed the Octopus documentation for deploying to Windows Azure
> http://docs.octopusdeploy.com/display/OD/Windows+Azure. Basically, you
> must install a tentacle on the same machine that hosts the Octopus server.
> Octopus then deploys locally to the tentacle before uploading your package
> to azure blob storage. My deploy.ps1
> https://github.com/davidpeden3/octopus-azure-transforms/blob/master/source/Deployment/deploy.ps1
> script performs the re-packaging post-transforms (executed by Octopus as
> configured here
> https://raw.githubusercontent.com/davidpeden3/octopus-azure-transforms/master/documentation/octopus-process.png)
> before the re-written package is published to Azure.
> 
> In other words, I simply leverage the deploy.ps1 _hook_ to inject my
> repackaging step into the native Octopus Azure deployment pipeline.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/davidpeden3/octopus-azure-transforms/issues/1#issuecomment-103752022
> .

## 

Kind Regards,

Igor Goldobin
Microsoft Certified Professional Developer
MInfTech, BEng, BIT, MCPD
Custom Web Design, IT Consulting, Enterprise Systems
Phone: 0409 096 917
Email:  igor.goldobin@gmail.com
Visit my website: www.e-fenix.com

**Comment by davidpeden3** [comment]
Exactly

**Comment by fenix2222** [comment]
Cool. Nice workaround by the way, I am setting it up now. I really wish
v3.0 was out already

On Wed, May 20, 2015 at 3:16 PM, David Peden notifications@github.com
wrote:

> Exactly
> 
> Sent from my iPhone
> 
> > On May 19, 2015, at 11:57 PM, fenix2222 notifications@github.com
> > wrote:
> > 
> > Hi David,
> > 
> > Does that mean that this screenshot is actually from "Deploy to Azure"
> > step
> > type? If so, does that mean that deploy.ps1 and postdeploy.ps1 will run
> > before actual octopus Azure Deployment script runs in that step? Just
> > trying to understand how this works
> > 
> > On Wed, May 20, 2015 at 2:43 PM, David Peden notifications@github.com
> > wrote:
> > 
> > > Hey @fenix2222 https://github.com/fenix2222,
> > > 
> > > First, I followed the Octopus documentation for deploying to Windows
> > > Azure
> > > http://docs.octopusdeploy.com/display/OD/Windows+Azure. Basically,
> > > you
> > > must install a tentacle on the same machine that hosts the Octopus
> > > server.
> > > Octopus then deploys locally to the tentacle before uploading your
> > > package
> > > to azure blob storage. My deploy.ps1
> > > <
> > > https://github.com/davidpeden3/octopus-azure-transforms/blob/master/source/Deployment/deploy.ps1
> > > 
> > > script performs the re-packaging post-transforms (executed by Octopus
> > > as
> > > configured here
> > > <
> > > https://raw.githubusercontent.com/davidpeden3/octopus-azure-transforms/master/documentation/octopus-process.png
> > > )
> > > before the re-written package is published to Azure.
> > > 
> > > In other words, I simply leverage the deploy.ps1 _hook_ to inject my
> > > repackaging step into the native Octopus Azure deployment pipeline.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > > https://github.com/davidpeden3/octopus-azure-transforms/issues/1#issuecomment-103752022
> > > 
> > > .
> > 
> > ## 
> > 
> > Kind Regards,
> > 
> > Igor Goldobin
> > Microsoft Certified Professional Developer
> > MInfTech, BEng, BIT, MCPD
> > Custom Web Design, IT Consulting, Enterprise Systems
> > Phone: 0409 096 917
> > Email: igor.goldobin@gmail.com
> > Visit my website: www.e-fenix.com
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/davidpeden3/octopus-azure-transforms/issues/1#issuecomment-103762410
> .

## 

Kind Regards,

Igor Goldobin
Microsoft Certified Professional Developer
MInfTech, BEng, BIT, MCPD
Custom Web Design, IT Consulting, Enterprise Systems
Phone: 0409 096 917
Email:  igor.goldobin@gmail.com
Visit my website: www.e-fenix.com
